pydoc-markdown/src/pydoc_markdown/contrib/renderers/markdown.py

```diff
@@ -205,7 +210,11 @@ def _render_toc(self, fp, level, obj):
     if level > self.toc_maxdepth:
       return
     object_id = self._resolver.generate_object_id(obj)
-    fp.write('  ' * level + '* [{}](#{})\n'.format(self._escape(obj.name), object_id))
+    title = self._escape(obj.name)
+    if not self.add_module_prefix and isinstance(obj, docspec.Module):
+      object_id = ".".join(object_id.split('.')[1:])
+      title = ".".join(title.split('.')[1:])
```

Comment:
I think you want to take the last element after the split, not "all but the first", right?

What you have no gives

```
my.module -> module
my.module.utils -> module.utils
```

but I feel like the prefix in the second example is not `my` but `my.module`.

Reply:
I will test it now with different configuration settings!
for my case, it's what I want but let me check thoroughly.

There is no tests right?

Reply:
Currently reworking the unit tests, which should make it easier for people to create new tests. If you merge `develop` into your branch, you can then add a new `.txt` file into the `pydoc-markdown/src/test/testcases/renderers/markdown` folder. Examples are here: https://github.com/NiklasRosenstein/pydoc-markdown/tree/develop/pydoc-markdown/src/test/testcases/renderers/markdown

We should definitely have a unit test for this option. :)

Reply:
> but I feel like the prefix in the second example is not `my` but `my.module`.

I see what you meant now! I'm not sure of the best approach!

Reply:
Using `".".join(title.split('.')[1:])`
![image](https://user-images.githubusercontent.com/7041726/128713059-65526e4b-e127-4047-8af3-f2d80e4bb65b.png)

Using `title.split('.')[-1]`
![image](https://user-images.githubusercontent.com/7041726/128713381-a73c90cc-4b50-4582-8d7b-16831e271c4c.png)


I'm not sure of what is more expected from the option! Your call 😋

Reply:
I think you are completely right: c616fa079001cc6fccdddb9223789f003b75c8e8

Reply:
👍 I'm also of the opinion that it should take only the last item after the split!

Reply:
wait shouldn't it be instead (indentation important):

- helpers
  - enumItemsFromList
  - addons
    - addon_check
  - append
  - ...
- ...

rather than the highly misleading:

- helpers
  - enumItemsFromList
- addons
  - addon_check
- append
- ...
- ...

Reply:
Yes, you are right, but indentation should be there only when all prefixes are removed right?
Because with the default settings it will be rather strange:

- mtb_blender.helpers
    - mtb_blender.helpers.addons
      - addon_check

Reply:
Good point @casperdcl , this is a bit confusing when documenting multiple modules in the same file and disabling the new `add_module_prefix` option. Currently in `docspec` (which pydoc-markdown uses), modules are handled non-recursively (so you get a flat list of module objects with fully qualified module names rather than a nested tree that includes modules), that's one reason where this comes from.

I was planning to add an option to docspec to keep the module hierarchy in tact, but will have to do it in a backwards compatible way. That would make the toc here render with the proper indentations (and the split "hack" from this PR would not be needed), but it will have some implications for how the processors/renderers work in pydoc-markdown.

Reply:
well I think either "option to remove module prefix" should also automatically "add indentation too"... or we also need to provide a separate "option to indent"

Reply:
The current state is still valid if you're only including a single module into one page. If you're willing to invest into the changes needed to fix the TOC indentation (I'd prefer to go with another option here, it could default to being enabled when `add_module_prefix` is turned off), I'm happy to review another PR. Otherwise, I'll keep it on my list for when docspec supports module hierarchy.